scanpy/preprocessing/_utils.py

```diff
@@ -1,14 +1,17 @@
 import numpy as np
-from scipy import sparse
 import numba
+from scipy import sparse
+
+from .._utils import _FlatSupportedArray, get_ufuncs
 
 
-def _get_mean_var(X, *, axis=0):
-    if sparse.issparse(X):
+def _get_mean_var(X: _FlatSupportedArray, *, axis=0):
+    if sparse.issparse(X):  # TODO: sparse-in-dask?
         mean, var = sparse_mean_variance_axis(X, axis=axis)
     else:
-        mean = np.mean(X, axis=axis, dtype=np.float64)
-        mean_sq = np.multiply(X, X).mean(axis=axis, dtype=np.float64)
+        ufuncs = get_ufuncs(X)
+        mean = ufuncs.mean(X, axis=axis, dtype=np.float64)
+        mean_sq = ufuncs.multiply(X, X).mean(axis=axis, dtype=np.float64)
```

Comment:
Instead of calling this `ufuncs`, can we pretend dask does array-api instead? I think that'll be closer to the truth, and be closer to a common pattern for interoperable packages.

```suggestion
        xp = get_array_module(X)
        mean = xp.mean(X, axis=axis, dtype=np.float64)
        mean_sq = xp.multiply(X, X).mean(axis=axis, dtype=np.float64)
```

Similar can be done above if we switch from `signbit` to `sign` (https://data-apis.org/array-api/2022.12/API_specification/generated/array_api.sign.html)

Reply:
Alternatively we can avoid all that and do:

```suggestion
        mean = X.mean(axis=axis, dtype=np.float64)
        mean_sq = (X * X).mean(axis=axis, dtype=np.float64)
```

Reply:
sadly not because `spmatrix * spmatrix` is a matrix multiplication

Reply:
ah, nice, because it’s in the non-sparse branch